crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -3443,6 +3449,21 @@ impl<
                 let sym_reg = storage_manager.load_to_general_reg(buf, &value);
                 ASM::mov_mem64_offset32_reg64(buf, ptr_reg, element_offset, sym_reg);
             }
+            Layout::LambdaSet(lambda_set) => {
+                let layout = layout_interner.get(lambda_set.runtime_representation());
+
+                Self::ptr_write(
+                    buf,
+                    storage_manager,
+                    layout_interner,
+                    ptr_reg,
+                    element_offset,
+                    element_width,
+                    layout,
+                    value,
+                );
+            }
```

Comment:
an additional fix I folded in to make sure this was not a problem. it is unrelated to the equality stuff, but totally straightforward: if we find a lambda set, recurse on the runtime representation.